presto-docs/src/main/sphinx/admin/function-namespace-managers.rst

```diff
@@ -0,0 +1,34 @@
+===========================
+Function Namespace Managers
+===========================
+
+.. warning::
+
+    This is an experimental feature being actively developed. The way
+    Function Namespace Managers are configured might be changed.
+
+Function namespace managers support storing and retrieving user-defined
+functions with a persistent storage, allowing users to create, alter,
+and delete functions, as well as to reference them within queries.
+
+Presto by default supports a MySQL-based function namespace manager.
+It is pluggable and can be configured by an administrator.
+
+To instantiate a MySQL-based function namespace manager with id
+``prod``, add a file ``etc/function-namespace/prod.properties``
+with the following contents::
+
+    function-namespace-manager.name=xdb
+    serving-catalogs=catalog1.schema1,catalog2
```

Comment:
I thought this is only configuring catalog, how does it work with `catalog1.schema1`?

Reply:
If you are planning to go with each function namespace manager will manage full catalog, just use the same model as connectors where the name of the file is the catalog name, and each config will manage a single catalog. Otherwise the naming is really confusing. Does the name of the config file mean anything?

Reply:
Moving to 1 catalog per FNM model, with the file name being the catalog name.